x/mock/test_utils.go

```diff
@@ -75,8 +75,9 @@ func SignCheckDeliver(
 	seq []uint64, expSimPass, expPass bool, priv ...crypto.PrivKey,
 ) sdk.Result {
 	tx := GenTx(msgs, accNums, seq, priv...)
+
 	// Must simulate now as CheckTx doesn't run Msgs anymore
-	res := app.Simulate(tx)
+	res := app.Simulate(nil, tx)
```

Comment:
Hmm shouldn't we simulate this with the actual serialized bytes?

Reply:
Indeed. Will amend.